keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1525 @@
+| 8.  | Unspecified  | Specified   | Specified    | Specified   | PL           | Specified PL | Specified CR | Specified CL |
+| 9.  | Specified    | Unspecified | Unspecified  | Unspecified | Specified PR | max          | Implementation defined min | max          |
+| 10. | Specified    | Unspecified | Unspecified  | Specified   | Specified PR | max or &sum;{CL} if all containers have limits | min |  Specified CL |
+| 11. | Specified    | Unspecified | Specified    | Unspecified | Specified PR | max          | Specified CR | max |
+| 12. | Specified    | Unspecified | Specified    | Specified   | Specified PR | max or &sum;{CL} if all containers have limits | Specified CR | Specified CL |
+| 13. | Specified    | Specified   | Unspecified  | Unspecified | Specified PR | Specified PL | Implementation defined min | PL (capping each container's limit by PL)|
+| 14. | Specified    | Specified   | Unspecified  | Specified   | Specified PR | Specified PL | Implementation defined min | Specified CL |
+| 15. | Specified    | Specified   | Specified    | Unspecified | Specified PR | Specified PL | Specified CR | PL (capping each container's limit by PL)|
+| 16. | Specified    | Specified   | Specified    | Specified   | Specified PR | Specified PL | Specified CR | Specified CL | 
+
+<br>
+
+##### Clarifying Tricky Cases and Design Decisions with Pod Level Resources
+There are two possible options to apply the default logic. 
+
+  **Option 1 (Rejected): Set unspecified Container Requests equal to Container
+  Limits, and derive unspecified Pod Requests from aggregated Container Requests**
+
+   * Defaulting container requests to container limits: If container requests are
+     unspecified, set them equal to container limits (which were just set to the
+     pod limit).
+   * Aggregating Container Requests: Calculate the pod level request by adding up
+     the requests of all the containers.
+   * Equalizing container limits with pod limit: When container level limits are
+     not set, each container's resource limit is set to match the pod's overall
+     limit.   
+
+  This approach can lead a situation where the calculated pod request (derived by
+  aggregating container requests) exceeds the defined-pod limit, which is 
+  logically inconsistent. Furthermore, this approach requires a multi-step
+  calculation to derive pod request that might be less intuitive to grasp. It
+  involes first setting container limits, then defaulting the container requests
+  to those limits, and finally aggregating those requests to determine pod
+  request.
+  
+  The examples in the next section will further illustrate the issue and how the
+  chosen approach (Option 2) provides a simpler alternative.
+
+  **Option 2 (Chosen): Set Pod Level Request equal to Pod Level Limit for Simplicity.**
+ 
+   * Defaulting pod request to pod limit: If the user doesn't explicitly specify
```

Comment:
so if pod level is set to 100, when there are two containers each asking for 60 request and limit, we are entering the "bad state" by defaulting pod level request to limit?

It is similar to @tallclair comment above I think

Reply:
This will be an invalid pod spec. There's a validation rule this kep proposes: "Sum of Container Level Requests is Never Greater Than Pod Level Request".

Reply:
hm, yes, I was thinking of something else. If total request is larger than the limit of the Pod - it does makes sense to make this pod invalid. I will think more about what confused me